integrations/cdi/oci/src/main/java/io/helidon/integrations/cdi/oci/OciExtension.java

```diff
@@ -0,0 +1,1570 @@
+                }
+                throw new IllegalStateException(exception.getMessage(), exception);
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> void fire(Instance<? super Object> instance, T payload, Annotation[] qualifiers) {
+        instance.select(EVENT_OBJECT_TYPE_LITERAL).get().select((Class<T>) payload.getClass(), qualifiers).fire(payload);
+    }
+
+    private static boolean isUnsatisfied(BeanManager bm, TypeAndQualifiers taq) {
+        return isUnsatisfied(bm, taq.type(), taq.qualifiers());
+    }
+
+    private static boolean isUnsatisfied(BeanManager bm, Type type, Annotation[] qualifiers) {
+        try {
+            return bm.resolve(bm.getBeans(type, qualifiers)) == null;
+        } catch (AmbiguousResolutionException ambiguousResolutionException) {
+            return false;
+        }
+    }
+
+    private static boolean equals(ProtectionDomain pd0, ProtectionDomain pd1) {
+        if (pd0 == null) {
+            return pd1 == null;
+        } else if (pd1 == null) {
+            return false;
+        }
+        return equals(pd0.getCodeSource(), pd1.getCodeSource());
+    }
+
+    private static boolean equals(CodeSource cs0, CodeSource cs1) {
+        if (cs0 == null) {
+            return cs1 == null;
+        } else if (cs1 == null) {
+            return false;
+        }
+        return equals(cs0.getLocation(), cs1.getLocation());
+    }
+
+    private static boolean equals(URL url0, URL url1) {
+        if (url0 == null) {
+            return url1 == null;
+        } else if (url1 == null) {
+            return false;
+        }
+        try {
+            return Objects.equals(url0.toURI(), url1.toURI());
+        } catch (URISyntaxException uriSyntaxException) {
+            // Use URL#equals(Object) only as a last resort, since it
+            // involves DNS lookups (!).
+            return url0.equals(url1);
+        }
+    }
+
+    private static Class<?> loadClassUnresolved(String name) throws ClassNotFoundException {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        return Class.forName(name, false, cl == null ? OciExtension.class.getClassLoader() : cl);
+    }
+
+    private static void customizeEndpointResolution(ClientBuilderBase<?, ?> clientBuilder) {
+        EndpointAdjuster ea = EndpointAdjuster.of(clientBuilder.getClass().getName());
+        if (ea != null) {
+            clientBuilder.additionalClientConfigurator(ea);
+        }
+    }
+
+
+    /*
+     * Inner and nested classes.
+     */
+
+
+    private static class TypeAndQualifiers {
+
+
+        /*
+         * Instance fields.
+         */
+
+
+        private final Type type;
+
+        private final Annotation[] qualifiers;
+
+        private final int hashCode;
+
+
+        /*
+         * Constructors.
+         */
+
+
+        private TypeAndQualifiers(final Type type, final Annotation[] qualifiers) {
+            super();
```

Comment:
Why? :thinking:

Reply:
Why the `super()` call? I like being explicit, I guess…